net/net.go

```diff
@@ -132,3 +135,23 @@ func IsIPv6CIDR(cidr *net.IPNet) bool {
 	ip := cidr.IP
 	return IsIPv6(ip)
 }
+
+// GetHostname returns OS's hostname if 'hostnameOverride' is empty; otherwise, return 'hostnameOverride'
+func GetHostname(hostnameOverride string) (string, error) {
```

Comment:
@thockin - I'm not sure this should get in `utils` as is. we either should drop the override behaviour or rename the func to show the real intent

Reply:
@yastij dropping override behaviour would make it almost identical to os.Hostname. Let's rename it. Any name suggestions?

Reply:
dering to @thockin on this